posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc

```diff
@@ -0,0 +1,412 @@
+
+A common use case for serverless applications is to process events coming from non-HTTP sources, such as Kafka topics or IBM Cloud Object Storage. Historically, to consume these events the application had to make use of the cloud provider's SDK, which locks the application into that provider. This is where CloudEvents comes in, as it provides a cross-provider standard around which data can be transmitted and received by an application, improving portability and reducing large dependencies.
+
+
+=== Updating the application to use CloudEvents
+IBM Cloud Code Engine connects IBM Cloud event providers such as https://cloud.ibm.com/docs/codeengine?topic=codeengine-working-kafkaevent-producer#setup-kafka-receiverapp[Kafka] or https://cloud.ibm.com/docs/codeengine?topic=codeengine-eventing-cosevent-producer#obstorage_ev_app[IBM Cloud Object Storage] and the application by using CloudEvents. These events are sent to a subscribed application as HTTP POST requests. Because the requests are in HTTP format, a RESTful application can receive these events without needing additional libraries and configuration.
+
+To include the CloudEvents `restful-ws` library within the application, Add the following https://openliberty.io/guides/maven-intro.html[Maven] dependency to your `pom.xml`:
+
+```
+<dependency>
+    <groupId>io.cloudevents</groupId>
+    <artifactId>cloudevents-http-restful-ws-jakarta</artifactId>
+    <version>2.5.0</version>
+</dependency>
+```
+
+Or if you use https://openliberty.io/guides/gradle-intro.html[Gradle]:
+```
+dependencies {
+	implementation group: 'io.cloudevents', name: 'cloudevents-http-restful-ws', version: '2.5.0'}
+
+```
+
+Two `restful-ws` packages are provided for CloudEvents:
+
+* `restful-ws` supports Jakarta EE 8 and the `javax.*` namespace
+* `restful-ws-jakarta` supports Jakarta EE 9+ and the `jakarta.*` namespace*
+
+The CloudEvent 2.5.0 release, the first release of the `restful-ws-jakarta` package, is at the time of this writing  not  yet available. If you want to use the new package before its release, you can build the package locally and store it in a suitable Maven repository for use with your application build. If you build from source,  change the version  to `2.5.0-SNAPSHOT`. You can find the source at https://github.com/cloudevents/sdk-java/tree/master/http/restful-ws-jakarta[HTTP Protocol Binding for Jakarta EE9+ - Jakarta RESTful Web Services].
+
+When run within the context of Open Liberty, the `CloudEventsProvider` class is automatically configured for marshalling and unmarshalling of CloudEvents.
+
+With the library imported, you can now update the application to use CloudEvents.
+
+The following sections guide you through updating the `SystemResource` class from the Getting Started guide  to use CloudEvents. You can view the <<completed_class>> to check that changes you make align with the expected result.
```

Comment:
consider moving the example up here- since the blogs enable scrolling within a codeblock, it wont take up too much page space and having to scroll to the bottom over and over is not ideal. I think readers will want to refer to this example throughout the steps and the snippets can be tough to orient